api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentReportController.java

```diff
@@ -49,20 +48,24 @@ public PaymentReportController(CardPaymentsReportScheduler cardPaymentsReportSch
         @ApiResponse(code = 200, message = "Reports sent")
     })
     @RequestMapping(value = "/payments/email-pay-reports", method = POST)
-    public void retrievePayments(@RequestParam(name = "start_date", required = false) Optional<String> startDateString,
+    public void generateAndEmailReport(@RequestParam(name = "payment_method", required = false) String paymentMethodType,
+                                 @RequestParam(name = "service_name", required = false) String serviceName,
+                                 @RequestParam(name = "start_date", required = false) Optional<String> startDateString,
                                  @RequestParam(name = "end_date", required = false) Optional<String> endDateString) {
 
+        validator.validate(paymentMethodType, startDateString, endDateString);
+
         Date fromDate = startDateString.map(s -> clock.atStartOfDay(s, FORMATTER)).orElseGet(clock::getYesterdayDate);
         Date toDate = endDateString.map(s -> clock.atEndOfDay(s, FORMATTER)).orElseGet(clock::getTodayDate);
 
         if (cardReportsEnabled) {
-            cardPaymentsReportScheduler.generateCardPaymentsReportTask(fromDate, toDate);
+            paymentsReportService.generateCardPaymentsCsvAndSendEmail(fromDate, toDate, serviceName);
```

Comment:
will handle proper hooking based on paymentMethod etc as part of another related story PAY-1071